protocols/luci-proto-yggdrasil/root/usr/libexec/rpcd/luci.yggdrasil

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+. /usr/share/libubox/jshn.sh
+
+case "$1" in
+	list)
+		json_init
+		json_add_object "generateKeyPair"
+		json_close_object
+		json_add_object "getPeers"
+		json_add_string "interface"
+		json_close_object
+		json_dump
+	;;
+	call)
+		case "$2" in
+			generateKeyPair)
+				json_load "$(yggdrasil -genconf -json)"
+				json_get_vars PublicKey PrivateKey
+				json_cleanup
+				json_init
+				json_add_object "keys"
+				json_add_string "priv" "$PrivateKey"
+				json_add_string "pub" "$PublicKey"
+				json_close_object
+				json_dump
+			;;
+			getPeers)
+				read -r input
+				json_load "$input"
+				json_get_vars interface
+				echo "$(yggdrasilctl -endpoint=unix:///tmp/yggdrasil/${interface}.sock -json getPeers)"
```

Comment:
The `interface` parameter is a root shell injection, please quote the URL string where the variable is interpolated in.

Reply:
ok thanks good eye
tested with quoted url and removed the call to /bin/echo

Reply:
the code now reads:
```bash
...
getPeers)
  read -r input
  json_load "$input"
  json_get_vars interface
  yggdrasilctl -endpoint="unix:///tmp/yggdrasil/${interface}.sock" -json getPeers
;;
```

this should suffice? interface is eg; yggdrasil-tun2 or what is defined in uci/network  
ie: `yggdrasil-<interface>`

Reply:
is this change trash? sorry, please advise.